cpp/include/cudf/io/json.hpp

```diff
@@ -507,6 +523,18 @@ class json_reader_options_builder {
     return *this;
   }
 
+  /**
+   * @brief Set delimiter separating records in JSON lines
+   *
+   * @param delimiter Delimiter separating records in JSON lines
+   * @return this for chaining
+   */
+  json_reader_options_builder& delimiter(char delimiter)
+  {
+    options._delimiter = delimiter;
```

Comment:
This only complements the other suggestion
```suggestion
    options.set_delimiter(delimiter);
```